### PR TITLE
Fix for Postprocessing of Eigenvalues

### PR DESCRIPTION
**Pull request by philbucher**
Fix in case the ComputingModelPart != MainModelPart

### CODE REVIEW

**Comment by philbucher** [comment]
@roigcarlo do you know how Travis can be restarted?

**Comment by roigcarlo** [comment]
@philbucher yes, you can request a restart right in the build page (click details). Button is greyed out in the top right corner.

The probem is that today machines seem to be slower than normal or we have sensibly increment our build time in the trunk for some reason.

I've reactivated the tests for this branch but I'm not sure they will be able to finish in time :S

**Comment by philbucher** [comment]
@roigcarlo ok thx for the info

File: applications/StructuralMechanicsApplication/custom_utilities/eigenvector_to_solution_step_variable_transfer_utility.hpp [hunk 1]
```diff
@@ -88,6 +88,13 @@ class EigenvectorToSolutionStepVariableTransferUtility {
         {
             ModelPart::NodeType::DofsContainerType& rNodeDofs = itNode->GetDofs();
             Matrix& rNodeEigenvectors = itNode->GetValue(EIGENVECTOR_MATRIX);
+
+            // in case ComputingModelPart != MainModelPart:
+            if (rNodeDofs.size() != rNodeEigenvectors.size2())
```
Comment (RiccardoRossi) [inline applications/StructuralMechanicsApplication/custom_utilities/eigenvector_to_solution_step_variable_transfer_utility.hpp:93]: @philbucher 
i don't understand this check...

Riccardo

**Review by RiccardoRossi — APPROVED** [review]
I approve since it doesn't look harmful to me...still i don't understand what is being done

File: applications/StructuralMechanicsApplication/custom_utilities/eigenvector_to_solution_step_variable_transfer_utility.hpp (hunk 1 above)
Comment (msandre) [reply applications/StructuralMechanicsApplication/custom_utilities/eigenvector_to_solution_step_variable_transfer_utility.hpp:93]: We used this check a while ago to fix a problem with a structural model from Andreas. The eigensolver stores the eigen vectors in a dense matrix on the nodes https://github.com/KratosMultiphysics/Kratos/blob/master/applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/eigensolver_strategy.hpp#697. This only happens for nodes of the computing model part. The rest of the nodes in the model part return an empty matrix in this utility. Then it crashes due to invalid memory access. This check fixed the problem, but it leaves open the possibility that the row index is out of range. Another safety check is, it should only operate on the computing model part. I'm not sure if there's a way to enforce/check this.

**Comment by RiccardoRossi** [comment]
@philbucher @msandre i am merging to avoid overloading of the travis on monday

**Pull request merged**